### DISCUSSION TITLE
C examples in web-server repo?

### DISCUSSION THREAD

**Discussion by perp3rlabcom** [AxisCommunications/acap-native-sdk-examples#87] [Q&A] [answered]
Hi all, according to the [acap-native-sdk-examples/web-server/README.md](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/web-server#readme), some C examples are supposed to be in the app/ folder but I can't find any. Any clues on what's up with that?

**Comment by pataxis** [comment, marked as answer]
Hi @perp3rlabcom , thanks for your question and sorry for the delay in answer.

We made an improvement in the documentation in #90 , let us know if this solves your issue.

**Reply by perp3rlabcom** [reply]
Many thanks Patrik, no worries. Very good to update the documentation.

But still, I just want to read through the example code, not build or fiddle with docker images or anything like that at this point. Isn't the example code available on github somewhere?

(Also, all other examples I checked (like [axevent](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/axevent/send_event/app), [axoverlay](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/axoverlay/app), [hello-world](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/hello-world/app), [larod](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/larod/app) etc) have their code in the local repo's `app/` folder. Why is the `web-server` example different? No big deal, just confusing, at least from an outsider's perspective.)

**Reply by pataxis** [reply]
Thanks for your feedback, it makes sense. I created a task for it but can't say exactly when it will happen.

When it comes to having code in `app/` directory, for `axevent`, `hello-world` etc. the source code in these cases are written by Axis and is therefore provided directly in the example. For web-server the [monkey](https://github.com/monkey/monkey) library is built and example source code is copied from this project.

To see the the example source code you can first build the example image and then copy the build content to host, as is done in many of the other examples:
```sh
# Build the example image
docker build . --build-arg ARCH --tag web-server:$ARCH

# Copy the build content to host directory
docker cp $(docker create web-server:$ARCH):/opt/monkey/examples ./build-web-server
cd build-web-server
ls -l
```

**Reply by perp3rlabcom** [reply]
OK, I see. Yeah, this web server stuff is a somewhat different beast in more ways... Thanks for replies and suggestions!
